flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/config/MySqlSourceOptions.java

```diff
@@ -68,8 +68,10 @@ public class MySqlSourceOptions {
     public static final ConfigOption<String> SERVER_TIME_ZONE =
             ConfigOptions.key("server-time-zone")
                     .stringType()
-                    .defaultValue("UTC")
-                    .withDescription("The session time zone in database server.");
+                    .noDefaultValue()
+                    .withDescription(
+                            "The session time zone in database server. If not set, then "
```

Comment:
I've considered this pain spot too, about the inferred default timezone, how about use server's timezone as default value firstly and then use the current system timezone. Of course that user's configuration own the most high level priority.

Reply:
we can use `SELECT @@GLOBAL.time_zone;` statement to query the server timezone
https://dev.mysql.com/doc/refman/8.0/en/time-zone-support.html

Reply:
Hi @leonardBang , thanks for the insightful comment, and I am looking into this issue but have some doubts about this solution.

One thing is that, as the `time_zone` returned by MySQL could be an offset like '+8:00', there are lots of different timezone names matching this offset like 'Asia/Chongqing', 'Asia/Singapore', 'Australia/Perth', etc.  If we arbitrarily choose a timezone name matching the offset, we may miss some traits like 'Daylight saving time' or other local time adjustments. Also, 'CST' could mean 'China Standard Time' or 'Central Standard Time' in the US, so the time_zone value returned by MySQL might not be enough to accurately determine this value.

Also, in some cases, the client might not be able to directly connect to the MySQL server during Logical Plan generation, thus if we make active probes to the remote server at that time, Flink may get stuck for a long time, which is not friendly to users. Alternatively, if we move the timezone detection logic to the runtime, this is not trivial work anyway.

What do you think about these? Thank you again for the review : )

Reply:
@kylemeow Thanks for sharing your opinions, Happy to see you've considered the timezone name and DST mechanism.  Firstly, We can use `SELECT TIMEDIFF(NOW(), UTC_TIMESTAMP)` to get the  accurate timezone offset which has been considered the  'CST' that may mean 'China Standard Time' or 'Central Standard issue, so CST is not a problem here. 

About DST issue, as the returned offset may change due to the DST exchange, the client need to know the adjust the change dynamically, but unfortunately we(flink cdc or flink) only load the timezone or time offset once, thus the timestamp conversion would be wrong after the DST coming or leaving. Although MySQL deals temporal function and timestamp wrongly under DST zones until `8.0.26` IIUC [1],  it's still an issue in the long term. In a word, query server timezone might get wrong result if MySQL server's timezone used DST. 

Using system's default server timezone also exists some issue if user's database and flink cluster are deployed in different timezone. 

>MySQL server during Logical Plan generation

We've visited MySQL server  in compile phase(validator), it is not a new introducing issue.
[1] https://bugs.mysql.com/bug.php?id=94257

Reply:
@kylemeow Just like we have two poisons, one is wrong result by DST, the other is wrong result by  system's default timezone. Let's drink the later one, maybe we can reduce the influence by add validation for the MySQL server  and Flink Cluster are deployed under different timezone.

Reply:
Thanks @leonardBang for the detailed explanation. As for the 'Logical Plan generation' part, sorry for the ambiguity, I originally meant that when `MySqlTableSourceFactory` generates timezone settings and Debezium properties during Flink logical plan generation process, the client-side JVM might not be able to directly connect to the MySQL server to execute SQL queries.

But anyway, users must guarantee that MySQL server is accessible at runtime, thus we could do the timezone verification at runtime, e.g. in `com.ververica.cdc.debezium.DebeziumSourceFunction#open`, and when we find there is a difference between the inferred timezone and actual MySQL timezone, we could print a warning message or throw an exception.

Reply:
Using system's timezone as default timezone and validating it during runtime makes sense to me, An exception is better as it will lead to the data correctness.

Reply:
Added runtime timezone check as recommended : )